### PR TITLE
[KBFS-2892] TLF Resets in the Row

### PR DESCRIPTION
**Pull request by jzila**
This PR isn't yet useful on its own, but it sets up the scaffolding to support TLF resets in the Files tab. The next step is to implement the banners and allow the user to do something about the reset.

### CODE REVIEW

File: shared/actions/fs/index.js [hunk 1]
```diff
@@ -320,6 +322,74 @@ const loadMimeTypeResult = (mimeType: string, action: FsGen.MimeTypeLoadPayload)
     })
   )
 
+const inboxQuery = {
```
Comment (chrisnojima) [inline shared/actions/fs/index.js]: we actually moved this into a helper , i think @akalin did this

File: shared/actions/fs/index.js [hunk 2]
```diff
@@ -320,6 +322,74 @@ const loadMimeTypeResult = (mimeType: string, action: FsGen.MimeTypeLoadPayload)
...
+  const resetRpc = new EngineRpc.EngineRpcCall(
+    {
+      'chat.1.chatUi.chatInboxUnverified': function*({
+        inbox,
+      }: RPCChatTypes.ChatUiChatInboxUnverifiedRpcParam) {
+        const result: RPCChatTypes.UnverifiedInboxUIItems = JSON.parse(inbox)
+        // whatever
+        if (!result || !result.items) return EngineRpc.rpcResult()
+        const tlfs: Array<[Types.Path, Types.ResetMetadata]> = result.items.reduce((filtered, item: RPCChatTypes.UnverifiedInboxUIItem) => {
+          const visibility = item.visibility === 2 // private
```
Comment (chrisnojima) [inline shared/actions/fs/index.js]: there's generated constants for all these numbers

File: shared/actions/fs/index.js [hunk 3]
```diff
@@ -320,6 +322,74 @@ const loadMimeTypeResult = (mimeType: string, action: FsGen.MimeTypeLoadPayload)
...
+
+function* loadResets(action: FsGen.LoadResetsPayload): Saga.SagaGenerator<any, any> {
+  // TODO: maybe uncomment?
+  // const conversations = yield Saga.call(
+  //   RpcChatTypes.localFindConversationsLocalRpcPromise,
+  //   action.payload.tlfName
+  // )
+  const resetRpc = new EngineRpc.EngineRpcCall(
+    {
+      'chat.1.chatUi.chatInboxUnverified': function*({
```
Comment (chrisnojima) [inline shared/actions/fs/index.js]: do you really want only the unverified payload?

File: shared/reducers/fs.js [hunk 4]
```diff
@@ -64,9 +65,9 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
 
         return [
           path,
-          // Since both `folderListLoaded` and `favoritesLoaded` can change
-          // `pathItems`, we need to make sure that neither one clobbers the
-          // other's work.
+          // Since `folderListLoaded`, `favoritesLoaded`, and `loadResetsResult`
+          // can change `pathItems`, we need to make sure that neither one
+          // clobbers the others' work.
```
Comment (chrisnojima) [inline shared/reducers/fs.js]: we have a helper in chat that does this and makes it more clear, might be worth doing here, something like updatePathItem(existing, newItem) or something

File: shared/actions/fs/index.js (hunk 3 above)
Comment (jzila) [reply shared/actions/fs/index.js]: I'm not sure what the differences are here. I was modeling off of an example in the chat index.js file.

File: shared/reducers/fs.js (hunk 4 above)
Comment (jzila) [reply shared/reducers/fs.js]: I think I'm going to leave it for now, until we need it one more time.

**Comment by jzila** [comment]
Ready for another look.

File: shared/reducers/fs.js [hunk 5]
```diff
@@ -142,6 +140,22 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
...
+
+        return [
+          path,
+          // Since `folderListLoaded`, `favoritesLoaded`, and `loadResetsResult`
+          // can change `pathItems`, we need to make sure that neither one
+          // clobbers the others' work.
+          original.set('resetParticipants', item.resetParticipants),
+        ]
+      }, [])
+      const n = state.mergeIn(['pathItems'], resetsToMerge)
```
Comment (songgao) [inline shared/reducers/fs.js]: just return here as you did above?

File: shared/actions/fs/index.js [hunk 6]
```diff
@@ -320,6 +323,71 @@ const loadMimeTypeResult = (mimeType: string, action: FsGen.MimeTypeLoadPayload)
...
+
+const {team, impteamnative, impteamupgrade} = RPCChatTypes.commonConversationMembersType
+
+function* loadResets(action: FsGen.LoadResetsPayload): Saga.SagaGenerator<any, any> {
+  // TODO: maybe uncomment?
+  // const conversations = yield Saga.call(
+  //   RpcChatTypes.localFindConversationsLocalRpcPromise,
+  //   action.payload.tlfName
+  // )
+  const resetRpc = new EngineRpc.EngineRpcCall(
```
Comment (songgao) [inline shared/actions/fs/index.js]: How expensive is this call? We are calling it every time user enters a TLF list or a TLF; it seems getting the entire inbox may be an overkill. Not for this PR, but perhaps there can be a RPC for service to call, to notify us about TLF resets. WDYT?

Comment (jzila) [reply shared/actions/fs/index.js]: Yeah I agree, and that's what we do for the team badging. Long term it'd be better to do something like that.

**Comment by jzila** [comment]
@chrisnojima I'd appreciate a final review on this now that we're out of code freeze.

**Comment by jzila** [comment]
@keybase/react-hackers I need another review on this.

**Comment by jzila** [comment]
@chrisnojima ping

**Pull request merged**